src/main/java/org/opensrp/repository/postgres/SettingRepositoryImpl.java

```diff
@@ -0,0 +1,317 @@
+package org.opensrp.repository.postgres;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.UUID;
+
+import org.opensrp.domain.postgres.Settings;
+import org.opensrp.domain.postgres.SettingsMetadata;
+import org.opensrp.domain.postgres.SettingsMetadataExample;
+import org.opensrp.domain.setting.SettingConfiguration;
+import org.opensrp.repository.SettingRepository;
+import org.opensrp.repository.postgres.mapper.custom.CustomSettingMapper;
+import org.opensrp.repository.postgres.mapper.custom.CustomSettingMetadataMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository("settingRepositoryPostgres")
+public class SettingRepositoryImpl extends BaseRepositoryImpl<SettingConfiguration> implements SettingRepository {
+	
+	@Autowired
+	private CustomSettingMapper settingMapper;
+	
+	@Autowired
+	private CustomSettingMetadataMapper settingMetadataMapper;
+	
+	@Override
+	public SettingConfiguration get(String id) {
+		return convert(settingMetadataMapper.selectByDocumentId(id));
+	}
+	
+	public Settings addSetting(SettingConfiguration entity) {
+		org.opensrp.domain.postgres.Settings pgSetting = null;
+		
+		if (entity == null || entity.getIdentifier() == null) {
+			return pgSetting;
+		}
+		
+		if (retrievePrimaryKey(entity) != null) { // Setting already added
+			return pgSetting;
+		}
+		
+		if (entity.getId() == null)
+			entity.setId(UUID.randomUUID().toString());
```

Comment:
One format can be used for single-line conditionals.

I see braces are being used for the rest?